### PR TITLE
Encrypt / Decrypt UI

### LINKED ISSUE #149: Encrypt/decrypt endpoints are missing
I enabled the bootstrap setting `encrypt.key` for symmetric key encryption and I know it works on my system with Spring Boot `1.5.x` . However on JHipster 4.2 generated registry the endpoint does not get enabled and it returns a 404.
If there is another way or kind of encryption (asymmetric) that is being used, it is not documented. please advise.

Using jhipster-registry 3.0.1

### PR DESCRIPTION
**Pull request by PierreBesson**
fix #149

I have wanted to do this for a very long time...

screen shot 2017-06-26 at 17 52 50

screen shot 2017-06-26 at 17 57 56

Note that this doesn't work by default, as the `encrypt.key` property needs to be set for the `/encrypt` and `/decrypt`endpoints to be open.

### CODE REVIEW

File: src/main/webapp/app/layouts/navbar/navbar.component.html [hunk 1]
```diff
@@ -62,6 +62,12 @@
                         </a>
                     </li>
                     <li>
+                        <a class="dropdown-item" routerLinkActive="active" routerLink="encryption" (click)="collapseNavbar()">
```
Comment (deepu105) [inline src/main/webapp/app/layouts/navbar/navbar.component.html]: should this be a separate menu? can this be i any of existing menu?

File: src/main/webapp/app/registry/encryption/encryption.component.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    <div class="head">
+        <h2 class="d-inline-block">Encrypt and decrypt properties</h2>
+    </div>
+    <div class="panel panel-default">
+        <div class="panel-body">
+            <ngb-tabset type="pills" class="nav-fill">
+                <ngb-tab title="Encrypt" id="encrypt">
+                    <template ngbTabContent>
+                        <form>
+                            <textarea type="text" class="form-control" id="textToEncrypt"
```
Comment (deepu105) [inline src/main/webapp/app/registry/encryption/encryption.component.html]: add some top and bottom margin for the text area

File: src/main/webapp/app/registry/encryption/encryption.component.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
+<div>
+    <div class="head">
+        <h2 class="d-inline-block">Encrypt and decrypt properties</h2>
+    </div>
+    <div class="panel panel-default">
+        <div class="panel-body">
+            <ngb-tabset type="pills" class="nav-fill">
```
Comment (deepu105) [inline src/main/webapp/app/registry/encryption/encryption.component.html]: Make type=tabs as pills look weird

File: src/main/webapp/app/registry/encryption/encryption.component.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                        </form>
+                    </template>
+                </ngb-tab>
+            </ngb-tabset>
+
+            <div *ngIf="result">
+                <h3>Result:</h3>
+                <pre class="code-block">{{result}}</pre>
+            </div>
+            <jhi-alert-error>Please make sure the encryption key is setup correctly.</jhi-alert-error>
```
Comment (deepu105) [inline src/main/webapp/app/registry/encryption/encryption.component.html]: Does this work this was?

File: src/main/webapp/app/layouts/navbar/navbar.component.html (hunk 1 above)
Comment (PierreBesson) [inline src/main/webapp/app/layouts/navbar/navbar.component.html]: I don't see where else I can put this.

File: src/main/webapp/app/registry/encryption/encryption.component.html (hunk 3 above)
Comment (PierreBesson) [inline src/main/webapp/app/registry/encryption/encryption.component.html]: pills
screen shot 2017-06-26 at 18 44 01
tabs
screen shot 2017-06-26 at 18 44 15

I personally prefer pills as I don't know how to fix the styling for tabs...

File: src/main/webapp/app/registry/encryption/encryption.component.html (hunk 4 above)
Comment (PierreBesson) [inline src/main/webapp/app/registry/encryption/encryption.component.html]: No it does nothing. I'm removing it.

**Comment by abshkd** [comment]
This is awesome. thank you @PierreBesson

File: src/main/webapp/app/registry/encryption/encryption.component.html (hunk 3 above)
Comment (deepu105) [inline src/main/webapp/app/registry/encryption/encryption.component.html]: If you add padding to textarea below then tabs should look fine. Try it else leave it ill polish after merge

File: src/main/webapp/app/registry/encryption/encryption.component.html [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    <div class="head">
+        <h2 class="d-inline-block">Encrypt and decrypt properties</h2>
+    </div>
+    <div class="panel panel-default">
+        <div class="panel-body">
+            <ngb-tabset type="tabs" class="nav-fill">
+                <ngb-tab title="Encrypt" id="encrypt">
+                    <template ngbTabContent>
+                        <form>
+                            <textarea type="text" class="form-control" id="textToEncrypt"
```
Comment (deepu105) [inline src/main/webapp/app/registry/encryption/encryption.component.html]: Can you add `my-1` css class to the text area?

Comment (PierreBesson) [inline src/main/webapp/app/registry/encryption/encryption.component.html]: @deepu105 done. Thanks a lot for the review !
Is everything else OK for you ?

**Comment by deepu105** [comment]
Yes

On 27 Jun 2017 2:09 pm, "Pierre Besson" <notifications@github.com> wrote:

*@PierreBesson* commented on this pull request.
------------------------------

In src/main/webapp/app/registry/encryption/encryption.component.html
<https://github.com/jhipster/jhipster-registry/pull/157#discussion_r124254376>
:

> @@ -0,0 +1,66 @@
+<div>
+    <div class="head">
+        <h2 class="d-inline-block">Encrypt and decrypt properties</h2>
+    </div>
+    <div class="panel panel-default">
+        <div class="panel-body">
+            <ngb-tabset type="tabs" class="nav-fill">
+                <ngb-tab title="Encrypt" id="encrypt">
+                    <template ngbTabContent>
+                        <form>
+                            <textarea type="text"
class="form-control" id="textToEncrypt"

@deepu105 <https://github.com/deepu105> done. Thanks a lot for the review !
Is everything else OK for you ?

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/jhipster/jhipster-registry/pull/157#discussion_r124254376>,
or mute the thread

**Comment by pascalgrimaud** [comment]
Should it be merge directly into master ? Or should we follow our previous follow, with dev branch ?

**Comment by jdubois** [comment]
Normally we use the dev branch, as the master branch is used directly by the Heroku button (so it might break the Heroku button).
But if that's a simple change don't worry too much.

**Comment by PierreBesson** [comment]
Yes and I'm not the only one guilty or this @jdubois. 😉

**Comment by PierreBesson** [comment]
This PR is replaced by https://github.com/jhipster/jhipster-registry/pull/160 that is done on the correct "develop" branch.

**Pull request closed without merging**